src/praisonai-agents/test_advanced_tools.py

```
@@ -0,0 +1,269 @@
+        assert 'internet_search' in TOOL_MAPPINGS, "internet_search tool missing"
+        
+        print("✅ Backward compatibility maintained")
+        return True
+    except Exception as e:
+        print(f"❌ Backward compatibility error: {e}")
+        return False
+
+def test_comprehensive_example():
+    """Test a comprehensive tool with multiple features."""
+    try:
+        from praisonaiagents.tools import tool, cache, Field, Choice, ToolContext, Priority
```

Ruff (0.12.2)

`praisonaiagents.tools.cache` imported but unused

Remove unused import: `praisonaiagents.tools.cache`

(F401)